service/tbc/tbcfork_test.go

```diff
@@ -853,9 +853,11 @@ func errorIsOneOf(err error, errs []error) bool {
 }
 
 func TestFork(t *testing.T) {
+	wg := sync.WaitGroup{}
 	ctx, cancel := context.WithCancel(context.Background())
 	defer func() {
 		cancel()
+		wg.Wait()
```

Comment:
Yikes I am not sure I love a wait in a deferred closure. I think we need to revisit entry and exit of this entire function. Pretty much by definition the wait should be at the very bottom of the test followed by the delete of the temp dir. Are you sure this is the root cause or is this a symptom that this works around?